kv/mem/store.go

```diff
@@ -56,16 +56,28 @@ func NewValueWithData(vers int, data []byte) kv.Value {
 }
 
 type value struct {
-	version int
-	data    []byte
+	version  int
+	revision int
+	data     []byte
 }
 
 func (v value) Version() int                      { return v.version }
 func (v value) Unmarshal(msg proto.Message) error { return proto.Unmarshal(v.data, msg) }
-func (v value) IsNewer(other kv.Value) bool       { return v.version > other.Version() }
+func (v value) IsNewer(other kv.Value) bool {
+	otherValue, ok := other.(*value)
+	if !ok {
+		return v.version > other.Version()
+	}
+	if v.revision == otherValue.revision {
```

Comment:
Should we just do `return v.revision > otherValue.revision` instead?

Reply:
nope, there are lots of places where mem.Value was just created with a value but no revision, when revision is 0 it should just compare version. External users should also not care about the revision as well which is why there is why this is not accessible  externally

Reply:
Hm it seems if the values are created by reading from the mem stores they should always have revision set. Are you saying people directly create `mem.Value` and mess it? But how do they update the versions in that case?

Reply:
The constructor of the mem.Value allows users to set version, people have been creating mem.Value in many tests